### PR TITLE
correct FullDetectorResponse.py for the reading of sparse parameter

### PR DESCRIPTION
**Pull request by GallegoSav**
Just a small correction as in order to read correctly the sparse parameter in the response file. I also added a option in case the SP parameter is empty in the file.

### CODE REVIEW

File: cosipy/response/FullDetectorResponse.py [hunk 1]
```diff
@@ -190,21 +190,24 @@ def _open_rsp(cls, filename, Spectrumfile=None,norm="Linear" ,single_pixel = Fal
...
-                elif key == "SP" and line[1]=="true" :
-                    sparse = True
-                    
-                elif key == "SP" and line[1]=="false" :
-                    sparse = False
+             
 		
                 elif key == "MS":
-                    sparse = bool(line[1])	
+                    if line[1] == "true" :
```
Comment (israelmcmc) [inline cosipy/response/FullDetectorResponse.py:207]: Was `bool('true')`/ `bool('false')`  failing? I'd had thought it should work. Or what was the issue?

File: cosipy/response/FullDetectorResponse.py [hunk 2]
```diff
@@ -190,21 +190,24 @@ def _open_rsp(cls, filename, Spectrumfile=None,norm="Linear" ,single_pixel = Fal
                 elif key == 'SA':
                     area_sim = float(line[1])
                     
-                elif key == "SP" and line[1]!="true" and line[1]!="false" :
-                    norm = str(line[1])
+                elif key == "SP" :
                     
+                    try :
```
Comment (israelmcmc) [inline cosipy/response/FullDetectorResponse.py:195]: If I understand this correctly, this is mostly so it is backwards compatible? Why do some files have an empty SP and some don't?

**Review by israelmcmc — COMMENTED** [review]
Thanks @GallegoSav . I think I can merge this quickly, I just wrote a couple questions to make sure I understood the code.

File: cosipy/response/FullDetectorResponse.py (hunk 1 above)
Comment (GallegoSav) [reply cosipy/response/FullDetectorResponse.py:207]: Yeah me too but it is actually always return `True` , even bool("false")

File: cosipy/response/FullDetectorResponse.py (hunk 2 above)
Comment (GallegoSav) [reply cosipy/response/FullDetectorResponse.py:195]: I think Andreas said he used a new(old) version or maybe this is a bug. Anyway this line is more to check if it was empty or not and warn the user

File: cosipy/response/FullDetectorResponse.py (hunk 1 above)
Comment (israelmcmc) [reply cosipy/response/FullDetectorResponse.py:207]: Right! I thought I had checked, but saw it wrong. Never mind, thanks.

File: cosipy/response/FullDetectorResponse.py (hunk 2 above)
Comment (israelmcmc) [reply cosipy/response/FullDetectorResponse.py:195]: Ok, thanks.

**Comment by israelmcmc** [comment]
This looks good, I'll merge it. Thanks, @GallegoSav

**Review by israelmcmc — APPROVED** [review]

**Pull request merged**